pkg/ddl/partition.go

```diff
@@ -1833,6 +1833,26 @@ func checkPartitionFuncType(ctx sessionctx.Context, expr ast.ExprNode, schema st
 	return errors.Trace(dbterror.ErrPartitionFuncNotAllowed.GenWithStackByArgs("PARTITION"))
 }
 
+func checkPartitionFuncTypeExprString(ctx sessionctx.Context, expr, schema string, tblInfo *model.TableInfo) error {
```

Comment:
We could combine two functions and use type `any` for expr param.

Reply:
Yes, but I think the result would be less readable, since it needs to check the input type twice, once for converting to an expression.Expression and once for checking if it is an column, to give a more precise error.

If the two functions are small as they are, same name prefix and close to each other, is that OK? I can merge them, but the result will be schema check and `type == int` check to be common, while expression and error refinement will be different.

What do you prefer?